airflow/providers/amazon/aws/operators/step_function.py

```diff
@@ -74,9 +88,25 @@ def execute(self, context: Context):
             raise AirflowException(f"Failed to start State Machine execution for: {self.state_machine_arn}")
 
         self.log.info("Started State Machine execution for %s: %s", self.state_machine_arn, execution_arn)
-
+        if self.deferrable:
+            self.defer(
+                trigger=StepFunctionsStartExecutionTrigger(
+                    execution_arn=execution_arn,
+                    waiter_delay=self.waiter_delay,
+                    waiter_max_attempts=self.waiter_max_attempts,
+                    aws_conn_id=self.aws_conn_id,
+                    region_name=self.region_name,
+                ),
+                method_name="execute_complete",
+                timeout=timedelta(seconds=self.waiter_max_attempts * self.waiter_delay),
+            )
         return execution_arn
 
+    def execute_complete(self, context: Context, event: dict[str, Any] | None = None) -> None:
+        if event and event["status"] == "success":
```

Comment:
I don't really know in which case event could be None here, but I think it's not supposed to happen ? So we shouldn't plan around it, we'd rather have it blow ?

Reply:
hmm ok, I'd prefer the opposite then 
```
if not event or event["status"] != "success":
    raise
```
here if it's none, nothing happens

Reply:
I've been doing 
```
if event is None:
    self.log.error("Trigger error: event is None")
    raise AirflowException("Trigger error: event is None")
```
Although it should never be `None`, shouldn't we be explicit about what is happening. In almost all cases, `event` should never be `None`, but if someone either accidentally or on purpose sets it to `None` in the Trigger, we should raise an error specific to that case no?

Reply:
if you just `raise AirflowException(f"Trigger error: event is {event}")` you have both cases handled at once.